src/main/java/seedu/hippocampus/logic/parser/ParserUtil.java

```diff
@@ -104,6 +105,26 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code String birthday} into an {@code Birthday}.
+     * Returns default empty birthday if birthday is not specified.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code birthday} is invalid.
+     */
+    public static Birthday parseBirthday(String birthday) throws ParseException {
+        requireNonNull(birthday);
+        String trimmedBirthday = birthday.trim();
+        if (birthday == Birthday.EMPTY_BIRTHDAY_STRING) {
+            return Birthday.EMPTY_BIRTHDAY;
+        }
+        try {
+            return new Birthday(trimmedBirthday);
+        } catch (Exception e) {
```

Comment:
It might be best not to catch all exceptions here, perhaps writing the specific ones which might be thrown by the Birthday constructor?

Reply:
Probably an extension of using string validation, difficult to predict all possible user errors.

Reply:
Separately caught IllegalArgumentException (for invalid year) and DateTimeException (for invalid format) in latest commit.